cmd/script/api/payloadtype.go

```diff
@@ -965,6 +967,115 @@ func RegisterCRCCloseProposalHashType(L *lua.LState) {
 	L.SetField(mt, "__index", L.SetFuncs(L.NewTable(), crcProposalMethods))
 }
 
+// Constructor
+func newSecretaryGeneralProposal(L *lua.LState) int {
+	fmt.Println("newSecretaryGeneralProposal begin")
+
+	ownerPublicKeyStr := L.ToString(1)
+	ownerPrivateKeyStr := L.ToString(2)
+
+	proposalType := L.ToInt64(3)
+	draftHashStr := L.ToString(4)
+
+	secretaryGeneralPublicKeyStr := L.ToString(5)
+	secretaryGeneralPrivateKeyStr := L.ToString(6)
+	client, err := checkClient(L, 7)
+
+	fmt.Println("ownerPublicKeyStr", ownerPublicKeyStr)
+	fmt.Println("ownerPrivateKeyStr", ownerPrivateKeyStr)
+	fmt.Println("proposalType", proposalType)
+	fmt.Println("draftHashStr", draftHashStr)
+	fmt.Println("secretaryGeneralPublicKeyStr", secretaryGeneralPublicKeyStr)
+	fmt.Println("secretaryGeneralPrivateKeyStr", secretaryGeneralPrivateKeyStr)
+
+	secretaryGeneralPublicKey, _ := common.HexStringToBytes(secretaryGeneralPublicKeyStr)
+	ownPublicKey, _ := common.HexStringToBytes(ownerPublicKeyStr)
+	ownerPrivateKey, _ := common.HexStringToBytes(ownerPrivateKeyStr)
+	secretaryGeneralPrivateKey, _ := common.HexStringToBytes(secretaryGeneralPrivateKeyStr)
+
+	SecretaryGeneralDID, _ := getDiDFromPublicKey(secretaryGeneralPublicKey)
+	needSign := true
+	if err != nil {
+		needSign = false
+	}
+	draftHash, err := common.Uint256FromHexString(draftHashStr)
+	if err != nil {
+		fmt.Println("wrong draft proposal hash")
+		os.Exit(1)
+	}
+
+	account := client.GetMainAccount()
+	CRCouncilMembercode := account.RedeemScript
+	CRCouncilMemberDID, _ := getDIDFromCode(CRCouncilMembercode)
+
+	fmt.Printf("account %+v\n", account)
```

Comment:
Remove debug fmt.Printf

Reply:
This is meant for testing and I don't think they need to be modified

Reply:
This is meant for testing, and I don't think they need to be modified here